runhouse/servers/cluster_servlet.py

```diff
@@ -79,7 +79,7 @@ async def __init__(
 
         # Only send for clusters that have den_auth enabled and if we are logged in with a user's token
         # to authenticate the request
-        if self.cluster_config.get("den_auth", False) and configs.token:
+        if self.cluster_config.get("den_auth", False):
```

Comment:
can you add the check to see if `Path.exists` for the `~/.rh/config.yaml` I think you had this before and I switched it

Reply:
Also I don't get how `configs.token` wouldn't work as a check here -- wouldn't it pull from the backing file?

Reply:
@rohinb2: Up until now, when `cluster.up()` was called for the first time, the cluster servlet was initialed *before* the `config.yaml `was saved on the cluster. Therefore, both `Path.exists` for the` ~/.rh/config.yaml` and `configs.token` wouldn't work, because the `config.yaml` did not exist on the cluster at the time of check. 

I changed a bit the logic of the restart_server(), so the config.yaml will be created before spinning up / restarting the cluster servlert. I ensured it would be created only if it doesn't exist on the cluster. (Meaning it will be created only when we run `cluster.up()` on a new / terminated cluster <=> `restart_server()` runs for the first time. This way, if someone with access to my cluster will run `restart_server()` on my running cluster, his or her token will not override the token in the existing `config.yaml`. 

Any thoughts / objections?

Reply:
This sounds like a great plan!! I'll read through this more thoroughly